py/dml/expr.py

```diff
@@ -15,10 +15,11 @@
     'Expression',
     'NonValue',
     'mkLit', 'Lit',
-    'mkApply', 'Apply',
+    'mkApply', 'mkApplyInits', 'Apply',
```

Comment:
Naming: did you consider flipping the naming like `mkApply` for initializers vs `mkApplyExprArgs` or `mkApplyOld` for expressions, making initializers the new normal?

Reply:
I did a little, but not seriously. My thought is not only is `mkApply` still used for 1.2, it is also used _a lot_ internally, while `mkApplyInits` isn't. Plus, `mkApply` is simpler, and in a sense more primitive. It makes more sense to me for it to be the unqualified variant.

Reply:
That was my gut feeling too, until I actually grepped for it. 
The 1.2 use (int_register) is obscure and dying, so those don't count. Other than that, there's ~15 uses, of which you turn a handful into `mkApplyInits`, and more or less everything else is for calling stuff in dmllib.h. I always found the `mkApply(site, mkLit(site, 'blah', TFunction([...], TInt), [...])` syntax a bit clumsy, for just calling a dmllib function. (Also, the inarg typechecking added by the TFunction is usually redundant; it's usually guaranteed to be correct so entrusing gcc with that typecheck is more than acceptable.)

Maybe the longer-term path is to add a separate primitive for "call dmllib function"?

Reply:
Hmm... you are  right. I'd be convinced, but I'm torn on one last point: `mkApply`'s original role is the constructor for `Apply` which corresponds to a C application which works exclusively with expressions and not initializers. It could _maybe possibly_ be argued that having `mkApplyInits` be renamed to `mkApply` would bastardize the role of constructors, and, in fact, having the `mk-` prefix of `mkApplyInits` is already a mistake and it should be named something else like `codegen_apply`.